internal/aws/awsutil/conn.go

```diff
@@ -372,16 +373,43 @@ func getSTSRegionalEndpoint(r string) string {
 }
 
 func GetDefaultSession(logger *zap.Logger, cfg *AWSSessionSettings) (*session.Session, error) {
-	awsConfig := &aws.Config{
+	cfgFiles := getFallbackSharedConfigFiles(backwardsCompatibleUserHomeDir)
+	logger.Debug("Fallback shared config file(s)", zap.Strings("files", cfgFiles))
+	awsConfig := aws.Config{
 		Credentials: getRootCredentials(cfg),
 	}
-	result, serr := session.NewSession(awsConfig)
+	result, serr := session.NewSessionWithOptions(session.Options{
+		Config:            awsConfig,
+		SharedConfigFiles: cfgFiles,
+	})
 	if serr != nil {
 		logger.Error("Error in creating session object waiting 15 seconds", zap.Error(serr))
 		time.Sleep(15 * time.Second)
-		result, serr = session.NewSession(awsConfig)
+		result, serr = session.NewSessionWithOptions(session.Options{
+			Config:            awsConfig,
+			SharedConfigFiles: cfgFiles,
+		})
 		if serr != nil {
 			logger.Error("Retry failed for creating credential sessions", zap.Error(serr))
+			return result, serr
+		}
+	}
+	cred, err := result.Config.Credentials.Get()
+	if err != nil {
+		logger.Error("Failed to get credential from session", zap.Error(err))
```

Comment:
I think this should be info since this is not an error but expected behavior.

Reply:
This is an error. We should always be able to get credentials.